File: go/src/189.go
func reverse(nums []int) {
	length := len(nums)
	for i := 0; i < length/2; i++ {
		nums[i], nums[length-i-1] = nums[length-i-1], nums[i]
	}
}

func rotate(nums []int, k int) {
	if len(nums) == 0 {
		return
	}

	k %= len(nums)
	reverse(nums[:])
	reverse(nums[:k])
	reverse(nums[k:])
}

File: py3/189.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        if n <= 1:
            return
        
        ans = [0 for _ in range(n)]
        for i in range(n):
            ans[(i+k)%n]=nums[i]
        
        for i in range(n):
            nums[i] = ans[i]
        
        return